specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,289 @@
+
+Currently developers can delete the entire User Data Folder to clear this data. This has a few drawbacks: it removes the entire user data folder instead of specific parts which incurs performance costs later on, the WebView must be shutdown fully and then re-initialized, and deleting the entire User Data Folder is a complex process. 
+We are creating an API that will allow developers to clear the browsing data programatically in which the developer can specify the data kind to clear. 
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+
+# Description
+We expose browsing data clearing in two asynchronous APIs:
+```IDL
+HRESULT ClearBrowsingData(
+      [in] uint64 dataKinds,
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler); 
+
+HRESULT ClearBrowsingDataInTimeRange(
+      [in] uint64 dataKinds, 
+      [in] double startTime,
+      [in] double endTime, 
+      [in] ICoreWebView2ClearBrowsingDataCompletedHandler* handler);
+```
+The first method takes a uint64 parameter that consists of one or more COREWEBVIEW2_BROWSING_DATA_KIND passed in as well as a handler which will indicate if the proper data has been cleared successfully. The handler will respond with a `bool` indicating whether the option successfully cleared the intended data fully. This method clears the data for all time. 
+
+The second method takes the same parameters for the dataKinds and handler, as well as a start and end time in which the API should clear the corresponding data between. The time parameters correspond to how many seconds have passed since the UNIX epoch. 
+
+Both methods clear the data with the associated profile in which the method is called on.
+ 
+The browsing data kinds that are supported are listed below. These data kinds follow a hierarchical structure in which nested bullet points are included in their parent bullet point's data kind. 
+Ex: DOM storage is included in site data which is included in the profile data. Each of the following bullets correspond to a COREWEBVIEW2_BROWSING_DATA_KIND. 
+
+* Profile
+  * Site Data
+    * DOM Storage: App Cache, File Systems, Indexed DB, Local Storage, Web SQL, Cache Storage
+        Storage
+    * Cookies 
+  * HTTP Cache 
+  * Download History
+  * General Autofill 
+  * Password Autosave
+  * Browsing History
+  * Settings  
+
+# Examples
+
+## Win32 C++
+```cpp
+
+/// Function to clear the autofill data from the last hour
+bool EnvironmentComponent::ClearAutofillData() 
+{
+    wwil::com_ptr<ICoreWebView2> coreWebView2;
+    CHECK_FAILURE(m_controller->get_CoreWebView2(&coreWebView2));
+
+    auto webview7 = coreWebView2.try_query<ICoreWebView2_7>();
+    if (webview7)
+    {
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webview7->get_Profile(&profile));
+        double end_time = (double)std::time(nullptr);
```

Comment:
Here and elsewhere, our sample code follows Microsoft coding style which uses camel casing.
```suggestion
        double endTime = (double)std::time(nullptr);
```